tools/build_api.py

```diff
@@ -235,6 +236,8 @@ def find_valid_toolchain(target, toolchain):
         internal_tc_name = get_toolchain_name(target, toolchain_name)
         if toolchain == "ARM" and toolchain_name == "ARMC5" and index != 0:
             end_warnings.append(ARMC5_MIGRATION_WARNING)
+        elif toolchain_name == "uARM" or target.default_toolchain == "uARM":
```

Comment:
This is been addressed in the last commit

Reply:
Why was it removed there is no explanation that I can see?

Reply:
Instead of adding default toolchain check (line 237 and 239) in both if cases. I just moved into else part of ```if not TOOLCHAIN_CLASSES[internal_tc_name].check_executable(): ```  before return.